gluon-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -0,0 +1,99 @@
+		local other_ip="$(gluon-wan nslookup $gateway | grep -v $peer | grep 'Address [0-9]')"
+		if [ "$other_ip" == "" ]; then
+			exit 1
+		fi
+		echo "$(echo $other_ip | sed -r s/Address\ [0-9]://)$(echo $1 | egrep -oe :[0-9]+$)"
+		exit 0
+	fi
+	exit 1
+}
+
+# Do we already have a private-key? If not generate one
+temp=$(uci get wireguard.wireguard.privatekey)
+if [ "$?" -ne "0" ]; then
+	uci set wireguard.wireguard.privatekey=$(wg genkey)
+	uci commit wireguard
+fi
+
+# Is wireguard enabled?
+if [ "$(uci get wireguard.wireguard.enabled)" == "true" ] || [ "$(uci get wireguard.wireguard.enabled)" == "1" ]; then
+
+	#We assume we are not connected by default
+	CONNECTED=0
+
+	MESH_VPN_IFACE=$(uci get wireguard.wireguard.iface)
+
+	# Check connectivity to supernode
+	ping -c1 -w2  $(wg  | grep fe80 | cut -d'/' -f1 | awk '{print $3}')%$MESH_VPN_IFACE
+	if [ $? == 0 ]; then
+			CONNECTED=1
+	fi
+
+	# If we don't have a connection we try to connect
+	if [ $CONNECTED != 1 ]; then
+		gluon-wan /usr/sbin/ntpd -n -N -S /usr/sbin/ntpd-hotplug -p ntp.ext.ffmuc.net -q
+
+		# Get the number of configured peers and randomly select one
+		NUMBER_OF_PEERS=$(uci -q show wireguard | egrep -ce peer_[0-9]+.endpoint)
+		PEER=$(awk 'BEGIN{srand();print int(rand()*($NUMBER_OF_PEERS)+1) }')
+
+		endpoint="$(check_handshake "$(uci get wireguard.peer_$PEER.endpoint)")"
+
+		# Delete Interfaces
+		ip link set nomaster bat0 dev mesh-vpn &> /dev/null
+		ip link delete dev mesh-vpn &> /dev/null
+		ip link del $MESH_VPN_IFACE &> /dev/null
+		PUBLICKEY=$(uci get wireguard.wireguard.privatekey | wg pubkey)
+		SEGMENT=$(uci get gluon.core.domain)
+		
+                # For future use
+                gluon-wan wget -q  -O- --post-data='{"segment": "'"$SEGMENT"'","public_key": "'"$PUBLICKEY"'"}' http://gw03.ext.ffmuc.net:5000/api/v1/wg/key/exchange
+
+		# Bring up the wireguard interface
+		ip link add dev $MESH_VPN_IFACE type wireguard
+		uci get wireguard.wireguard.privatekey | wg set $MESH_VPN_IFACE private-key /proc/self/fd/0
+		ip link set up dev $MESH_VPN_IFACE
+
+		# Add link-address and Peer
+		ip address add "$(interface_linklocal "$MESH_VPN_IFACE")"/64 dev $MESH_VPN_IFACE
+		if [ "$endpoint" == "" ]; then
+			endpoint=$(uci get wireguard.peer_$PEER.endpoint)
+		fi
+		gluon-wan wg set $MESH_VPN_IFACE peer $(uci get wireguard.peer_$PEER.publickey) persistent-keepalive 25 allowed-ips $(uci get wireguard.peer_$PEER.link_address)/128 endpoint $endpoint
+		
+		# We need to allow incoming vxlan traffic on mesh iface
+		sleep 10
+        	ip6tables -I INPUT 1 -i $MESH_VPN_IFACE -m udp -p udp --dport 8472 -j ACCEPT
+		
+		# Bring up VXLAN
+		ip link add mesh-vpn type vxlan id "$(lua -e 'print(tonumber(require("gluon.util").domain_seed_bytes("gluon-mesh-vpn-vxlan", 3), 16))')" local $(interface_linklocal "$MESH_VPN_IFACE") remote $(uci get wireguard.peer_$PEER.link_address) dstport 8472 dev $MESH_VPN_IFACE
```

Comment:
This tries to create a VXLAN interface with the same name that is already used for Wireguard interface.

Reply:
No, we renamed the wg iface to wg_mesh_vpn

Reply:
Hm, strange for the wg interface is still created as mesh-vpn. Let me try a clean rebuild.

Reply:
After the clean rebuild of the latest commit the Wg interface is still created as mesh-vpn. Do you maybe have local changes not yet pushed? I don't the anything in the code that refers to wg_mesh_vpn.

Reply:
Maybe wrong interface?
```
# uci get wireguard.wireguard.iface
wg_mesh_vpn
```

Reply:
Ok, it was a misconfiguration. I created the site.conf a few days ago and all was working fine. I just noted that the interface name can be configured via the site.conf and it was set to mesh-vpn as stated in the example config of this package. Changing it to something else than mesh-vpn works.

It there any need to make the interface name configurable?

Reply:
I don't really know to be honest I just copied this from the old script, but I also think it causes more confusion than it actually helps.